lib/ClangImporter/ImportDecl.cpp

```diff
@@ -3200,9 +3200,13 @@ namespace {
         if (importedName.isSubscriptAccessor()) {
           assert(func->getParameters()->size() == 1);
           auto typeDecl = dc->getSelfNominalTypeDecl();
-          auto parameterType = func->getParameters()->get(0)->getType();
+          auto parameter = func->getParameters()->get(0);
+          auto parameterType = parameter->getType();
           if (!typeDecl || !parameterType)
             return nullptr;
+          if (parameter->isInOut())
+            // Subscripts with inout parameters are not allowed in Swift.
+            return nullptr;
```

Comment:
We should transform this to a non-inout parameter.

If you're not going to fix this in the next week or two please add a diagnostic here as well.

Reply:
Oh I missed what this is actually doing, this makes the other overload becomes usable. This is lower priority, then.

Reply:
It's quite challenging to add a useful diagnostic here, since this method only gets invoked from generated Swift code. I think it's not worth the trouble, but let's discuss if you disagree 😉